File: TS/oib-validator.ts
const isValidOIB = (oib:string): boolean => {
    if (!/^[0-9]{11}$/.test(oib)) 
        return false;

    let a = 10;
    for(const digit of oib.substring(0, 10)) {
        a += parseInt(digit);
        a %= 10;
        if (a === 0) 
            a = 10;
        a *= 2;
        a %= 11;
    }

    let kontrolni = 11 - a;
    if (kontrolni === 10) 
        kontrolni = 0;

    return kontrolni === parseInt(oib.substr(10, 1));
}

File: Dart/oib-validator.dart
 
  bool checkOIB(String oib) {
    if (oib.length != 11) return false;

    int a = 10;
    for (int i = 0; i < 10; i++) {
      int broj;
      try {
        broj = int.tryParse(oib.substring(i, i + 1)) ?? 0;
      } catch (e) {
        return false;
      }

      a += broj;
      a = a % 10;
      if (a == 0) a = 10;
      a *= 2;
      a = a % 11;
    }
    int kontrolni = 11 - a;
    if (kontrolni == 10) kontrolni = 0;

    int broj;

    try {
      broj = int.tryParse(oib.substring(10)) ?? 0;
    } catch (e) {
      return false;
    }
    return kontrolni == broj;
  }
